### PR TITLE
Add option to read Transformations from a file

### PR DESCRIPTION
**Pull request by okhasawn**
### Description
This PR adds the option (`--transformer-config-file`) for the Traffic Replayer to read the transformations from a file.

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/TrafficReplayer.java [hunk 1]
```diff
@@ -346,8 +354,23 @@ public static void main(String[] args)
...
+            if (params.transformerConfigFile != null && !params.transformerConfigFile.isBlank()) {
+                try {
+                    configContent = Files.readString(Paths.get(params.transformerConfigFile), StandardCharsets.UTF_8);
+                } catch (IOException e) {
+                    System.err.println("Error reading transformer configuration file: " + e.getMessage());
+                    System.exit(5);
+                }
+            } else if (params.transformerConfig != null && !params.transformerConfig.isBlank()) {
+                configContent = params.transformerConfig;
+            }
```
Comment (gregschohn) [inline TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/TrafficReplayer.java:371]: Please put these lines into a helper function to return the config (or null).  The helper function could throw or call exit() directly.  We'll do a rewrite of all of this parsing logic that's getting spread out in the near future.

**Review by gregschohn — COMMENTED** [review]
Once you refactor that, I'm fine with this change.  Thanks!

**Review by gregschohn — COMMENTED** [review]
Please also log the transformerConfig to info so that users will know what transformations were used for a given run.

**Comment by okhasawn** [comment]
Thanks for the comments @gregschohn!
All addressed in new commits.

**Comment by okhasawn** [comment]
I tested attempting #367 with the following `TransformerConfig.json` file:
```
[
  {"JsonJMESPathTransformerProvider": { "script": "{ \"settings\": settings, \"mappings\": payload.mappings.*.properties | [merge(@)] }"}}]
```

Results: can confirm that the above transformation does fix the bug. I was able to reproduce the bug without the transformation.

**Review by gregschohn — APPROVED** [review]
I'd like to revisit this whole file and refactor it.  I'm not sure exactly what this would look like after refactoring (specifically around error handling & System.exit()).  This is a much needed feature in the interim though.

**Pull request merged**